### DISCUSSION TITLE
How to change the mirror time zone to Asia/Shanghai but still be able to connect to MySQL from 429 onwards

### DISCUSSION THREAD

**Discussion by A-little-bit-of-data** [trinodb/trino#20047] [Q&A]
429 mysql connector starting, see the following explanation in the document: To preserve time instants, Trino sets the session time zone of the MySQL connection to match the JVM time zone. As a result, error messages similar to the following example occur when a timezone from the JVM does not exist on the MySQL server.
My MySQL time zone is CST. I change the time zone to Asia/Shanghai through dockerfile and use the modified image, and then connect to MySQL, and the following error will be reported:
1. Caused by: java.sql.SQLNonTransientConnectionException: Could not create connection to database server. Attempted reconnect 3 times. Giving up.
2. Caused by: com.mysql.cj.exceptions.CJException: Unknown or incorrect time zone: 'Asia/Shanghai'

The time zone is modified to Asia/Shanghai in order to output time + 8 in the log. In log.properties, I only see the setting level as INFO. There is no configuration that can set the log format in the document. How to solve this problem?

**Comment by terranger** [comment]
Try installing the missing time zone on the MySQL server, or add "-Duser.timezone=GMT+8" in the jvm.config file.

**Comment by A-little-bit-of-data** [comment]
> -Duser.timezone=GMT+8

Thank you very much for your answer. Generally, I cannot change the time zone of MySQL. I tried the second method and the results are as follows.
Caused by: IllegalArgumentException: The datetime zone id 'GMT+08:00' is not recognised

I tried using -Duser.timezone=UTC+8, but the time in the log is still UTC time

**Comment by terranger** [comment]
Sorry, or you can try using "-Duser.timezone=Etc/GMT-8" and see if it works.

**Comment by A-little-bit-of-data** [comment]
> Sorry, or you can try using "-Duser.timezone=Etc/GMT-8" and see if it works.

Thank you very much for your help, it solved my problem perfectly。

**Comment by A-little-bit-of-data** [comment]
> Sorry, or you can try using "-Duser.timezone=Etc/GMT-8" and see if it works.

I have another question and I would like to take a leave of absence. I don’t seem to see this configuration parameter in the document. Where can I generally find these configuration parameters?
